src/main/grammars/WebAssemblyLexer.flex

```diff
@@ -20,30 +20,123 @@ import org.jetbrains.webstorm.lang.psi.WebAssemblyTypes;
+    "align="                                        { return WebAssemblyTypes.ALIGNEQKEY; }
+    i (32 | 64) \.const                             { return WebAssemblyTypes.ICONST; }
+    f (32 | 64) \.const                             { return WebAssemblyTypes.FCONST; }
 
-<YYINITIAL> {KEY_CHARACTER}+                                { yybegin(YYINITIAL); return WebAssemblyTypes.KEY; }
+    "type"                                          { return WebAssemblyTypes.TYPEKEY; }
+    "import"                                        { return WebAssemblyTypes.IMPORTKEY; }
+    "table"                                         { return WebAssemblyTypes.TABLEKEY; }
+    "memory"                                        { return WebAssemblyTypes.MEMORYKEY; }
+    "global"                                        { return WebAssemblyTypes.GLOBALKEY; }
+    "local"                                         { return WebAssemblyTypes.LOCALKEY; }
+    "export"                                        { return WebAssemblyTypes.EXPORTKEY; }
+    "elem"                                          { return WebAssemblyTypes.ELEMKEY; }
+    "data"                                          { return WebAssemblyTypes.DATAKEY; }
+    "offset"                                        { return WebAssemblyTypes.OFFSETKEY; }
+    "start"                                         { return WebAssemblyTypes.STARTKEY; }
+    "module"                                        { return WebAssemblyTypes.MODULEKEY; }
+    "global"                                        { return WebAssemblyTypes.GLOBALKEY; }
 
-<YYINITIAL> {SEPARATOR}                                     { yybegin(WAITING_VALUE); return WebAssemblyTypes.SEPARATOR; }
+    "("                                             { return WebAssemblyTypes.LEFT_BRACKET; }
+    ")"                                             { return WebAssemblyTypes.RIGHT_BRACKET; }
 
-<WAITING_VALUE> {CRLF}({CRLF}|{WHITE_SPACE})+               { yybegin(YYINITIAL); return TokenType.WHITE_SPACE; }
+    {UN}                                            { return WebAssemblyTypes.UNSIGNED; }
+    {SN}                                            { return WebAssemblyTypes.SIGNED; }
+    {FN}                                            { return WebAssemblyTypes.FLOAT; }
+    {STRING}                                        { return WebAssemblyTypes.STRING; }
 
-<WAITING_VALUE> {WHITE_SPACE}+                              { yybegin(WAITING_VALUE); return TokenType.WHITE_SPACE; }
+    {ID}                                            { return WebAssemblyTypes.IDENTIFIER; }
 
-<WAITING_VALUE> {FIRST_VALUE_CHARACTER}{VALUE_CHARACTER}*   { yybegin(YYINITIAL); return WebAssemblyTypes.VALUE; }
+    {SPACE}+                                        { return TokenType.WHITE_SPACE; }
+    {LINECOMMENT}                                   { return WebAssemblyTypes.LINE_COMMENT; }
+    "(;"                                            { yybegin(BLOCKCHAR); return WebAssemblyTypes.COMMENT_LEFT_BR; }
```

Comment:
you should merge the comments to a single token, e.g. using merge lexer like in dart. See for example 
```
com.jetbrains.lang.dart.lexer.DartLexer#MERGE_FUNCTION
```


Also you have to add the comments to the COMMENTS collection in ParsingDefinition otherwise the parser doesn't understand the comment like real comments.